File: kata_11th_feb/pythonTasks/mystack.py
class MyStack:
    def __init__(self):
        self.is_empty = True
        self.size = 0
        self.elements = []

    def is_empty(self):
        self.is_empty = True
        return self.is_empty

    def push(self, element):
        for i in range(len(self.elements) - 1, -1, -1):
            self.elements[i] = element
        self.size += 1

    def is_not_empty(self):
        self.is_empty = False
        return self.is_empty

    def size(self):
        return self.size

    def get(self, index):
        return self.elements[index]

    def pop(self):
        element = self.elements[self.size - 1]
        self.elements[self.size - 1] = None
        self.size -= 1

    def search(self, element):
        for index in range(self.size):
            if self.elements[index] == element:
                return True
        return False

    def contains(self, element):
        return element in self.elements

    def peek(self):
        for index in range(self.size - 1):
            return self.elements[0]
        return None

File: kata_11th_feb/MyStack.js
class MyStack {
    constructor() {
        this.elements = new Array(6);
        this.size = 0;
    }

    isEmpty() {
        return this.size === 0;
    }

    push(element) {
        for (let i = this.elements.length - 1; i >= 0; i--) {
            this.elements[i] = element;
        }
        this.size++;
    }

    isNotEmpty() {
        return this.size > 0;
    }

    size() {
        return this.size;
    }

    get(index) {
        return this.elements[index];
    }

    pop() {
        if (this.size > 0) {
            let element = this.elements[--this.size];
            this.elements[this.size] = null;
            return element;
        }
        return null;
    }

    search(element) {
        for (let index = 0; index < this.size; index++) {
            if (this.elements[index] === element) {
                return true;
            }
        }
        return false;
    }

    contains(element) {
        for (let item of this.elements) {
            if (item === element) {
                return true;
            }
        }
        return false;
    }

    peek() {
        if (this.size > 0) {
            return this.elements[0];
        }
        return null;
    }
}
